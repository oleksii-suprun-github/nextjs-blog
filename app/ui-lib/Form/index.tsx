import { BaseSyntheticEvent, ReactNode } from 'react';

function Form({
  action,
  children,
}: {
  action: (e?: BaseSyntheticEvent) => Promise<void>;
  children: ReactNode;
}) {
  return (
    <form data-testid="form" onSubmit={action} className="flex flex-col text-gray-200">
      {children}
    </form>
  );
}

export default Form;
