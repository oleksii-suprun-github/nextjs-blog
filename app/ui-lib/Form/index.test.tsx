import { render, screen, fireEvent } from '@testing-library/react';
import Form from '.';

const mockAction = vi.fn().mockResolvedValue(undefined);

describe('Form Component', () => {
  test('renders children correctly', () => {
    render(
      <Form action={mockAction}>
        <div>Test Child</div>
      </Form>,
    );

    const childElement = screen.getByText('Test Child');
    expect(childElement).toBeInTheDocument();
  });

  test('calls action on form submit', async () => {
    render(
      <Form action={mockAction}>
        <button type="submit">Submit</button>
      </Form>,
    );

    const submitButton = screen.getByText('Submit');
    fireEvent.click(submitButton);

    expect(mockAction).toHaveBeenCalled();
  });

  test('applies correct styling', () => {
    render(
      <Form action={mockAction}>
        <div>Test Child</div>
      </Form>,
    );

    const formElement = screen.getByTestId('form');
    expect(formElement).toHaveClass('flex flex-col text-gray-200');
  });

  test('matches snapshot', () => {
    const { asFragment } = render(
      <Form action={mockAction}>
        <div>Test Child</div>
      </Form>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
