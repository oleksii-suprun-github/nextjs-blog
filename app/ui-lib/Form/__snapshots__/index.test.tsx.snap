// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Form Component > matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="flex flex-col text-gray-200"
    role="form"
  >
    <div>
      Test Child
    </div>
  </form>
</DocumentFragment>
`;
